src/context/client.rs

```diff
@@ -19,48 +20,92 @@ impl RedisModuleClientInfo {
 }
 
 /// GetClientNameById, GetClientUserNameById and GetClientCertificate use autoMemoryAdd on the ValkeyModuleString pointer
-// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
+/// after the callback (command, server event handler, ...) these ValkeyModuleString pointers will be freed automatically
 impl Context {
     pub fn get_client_id(&self) -> u64 {
         unsafe { RedisModule_GetClientId.unwrap()(self.ctx) }
     }
 
-    pub fn get_client_name(&self) -> ValkeyString {
-        let client_id = self.get_client_id();
+    /// wrapper for RedisModule_GetClientNameById
+    pub fn get_client_name_by_id(&self, client_id: u64) -> ValkeyString {
         let client_name = unsafe { RedisModule_GetClientNameById.unwrap()(self.ctx, client_id) };
```

Comment:
This API (VM_GetClientNameById) can return NULL if the client ID does not exist (if the client disconnected or if the Module was just providing an invalid client id).

Seems like we should handle this before calling the `ValkeyString::from_redis_module_string` function below.

We can probably return a Result here

Reply:
currently *_by_id functions return `"(NULL string reply referenced in module)"` if ID is invalid.  It could be a reasonable pattern for all *_by_id methods return ValkeyResult?

Reply:
updated the code to return ValkeyResult<ValkeyString> for all *by_id functions